datamol/reactions/_reactions.py

```diff
@@ -0,0 +1,109 @@
+from functools import singledispatch
+from typing import Union
+import numpy as np
+from rdkit.Chem import rdChemReactions
+from rdkit.Chem import AllChem
+
+import datamol as dm
+
+
+ATTACHING_RXN = rdChemReactions.ReactionFromSmarts("[*;h:1]>>[*:1][*]")
+
+
+def from_smarts(rxn_smarts: str):
+    """Create a reaction from smarts"""
+    return rdChemReactions.ReactionFromSmarts(rxn_smarts)
+
+
+def is_reaction_ok(rxn: Union[str, rdChemReactions.ChemicalReaction]):
+    """Check is the reaction is synthetically valid"""
+    if isinstance(rxn, str):
+        rxn = from_smarts(rxn_smarts=rxn)
+    return rdChemReactions.SanitizeRxn(rxn) == rdChemReactions.SanitizeFlags.SANITIZE_NONE
```

Comment:
check cases where this could be `None` or raise an exception on `rdChemReactions.SanitizeRxn`.